### PR TITLE
[mio-circle] Add guards

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise buffer_info() methods with assertion check and guards.

### CODE REVIEW

**Comment by seanshpark** [comment]
- for #12732
- from #13615

File: compiler/mio-circle08/src/Reader.cpp [hunk 1]
```diff
@@ -118,7 +120,10 @@ size_t Reader::buffer_info(uint32_t buf_idx, const uint8_t **buff_data, bool &ex
         return 0;
 
       ext_offset = true;
-      *buff_data = reinterpret_cast<const uint8_t *>(&_rawdata->at(buffer_offset));
+      if (buff_data != nullptr)
```
Comment (seanshpark) [inline compiler/mio-circle08/src/Reader.cpp:123]: some modules only want size and calls with null `buff_data`.

Comment (mhs4670go) [reply compiler/mio-circle08/src/Reader.cpp:123]: Then, if `buff_data` is `nullptr`, `buffer->size()` is zero?

Comment (seanshpark) [reply compiler/mio-circle08/src/Reader.cpp:123]: > Then, if buff_data is nullptr, buffer->size() is zero?

`buff_data` is return args, and if null, call doesn't want the buff_data ptr.
`buffer->size()` can have non zero value as it's size of `buffer`

this code change is same as below code for `else if (auto *array = buffer->data())` case.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**